packages/phone-number-privacy/signer/src/domain/domain.service.ts

```
@@ -110,4 +119,103 @@ export class DomainService implements IDomainService {
       )
     }
   }
+
+  public async handleGetBlindedMessagePartialSig(
+    request: Request<{}, {}, DomainRestrictedSignatureRequest>,
+    response: Response
+  ): Promise<void> {
+    const logger = response.locals.logger
+    const domain = request.body.domain
+    if (!this.authAndCheckKnownDomain(domain, response, Endpoints.DOMAIN_SIGN, logger)) {
+      logger.warning('Cannot handle request')
+      return
+    }
+    logger.info('Processing request to get domain signature ', {
+      name: domain.name,
+      version: domain.version,
+      hash: domainHash(domain),
+    })
+
+    const trx = await getTransaction()
+    const domainState = (await getDomainStateWithLock(domain, trx, logger))! // TODO: what if we dont have domain state yet
+    // @Alec we never actually create any domain state in first place.
+    const doesHaveRemainingQuota = await this.quotaService.doesHaveRemainingQuota(
+      domain,
+      domainState
+    )
+    if (!doesHaveRemainingQuota) {
+      trx.rollback()
+      logger.warn(`Exceeded quota`, {
+        name: domain.name,
+        version: domain.version,
+      })
+      // TODO: @Alec should I use here SignMessageResponseFailure?
+      respondWithError(Endpoints.DOMAIN_SIGN, response, 403, WarningMessage.EXCEEDED_QUOTA)
+      return
+    }
+
+    let signature: string
+    try {
+      const keyProvider = getKeyProvider()
+      const privateKey = keyProvider.getPrivateKey()
+      signature = computeBlindedSignature(request.body.blindedMessage, privateKey, logger)
+    } catch (err) {
+      trx.rollback()
+      throw err
+      // TODO @Alec throw error or create response? I used code from 'get-partial-signature.ts'
```

We should throw the error and then have the whole function in a try-catch where the catch responds with 500 (as we do in 'get-partial-signature.ts'